packages/eslint-plugin/src/rules/prefer-nullish-coalescing.ts

```diff
@@ -547,6 +515,76 @@ export default createRule<Options, MessageIds>({
           });
         }
       },
+      IfStatement(node: TSESTree.IfStatement): void {
+        if (node.alternate != null) {
+          return;
+        }
+
+        let assignmentExpression: TSESTree.Expression | undefined;
+        if (
+          node.consequent.type === AST_NODE_TYPES.BlockStatement &&
+          node.consequent.body.length === 1 &&
+          node.consequent.body[0].type === AST_NODE_TYPES.ExpressionStatement
+        ) {
+          assignmentExpression = node.consequent.body[0].expression;
+        } else if (
+          node.consequent.type === AST_NODE_TYPES.ExpressionStatement
+        ) {
+          assignmentExpression = node.consequent.expression;
+        }
+
+        if (
+          !assignmentExpression ||
+          assignmentExpression.type !== AST_NODE_TYPES.AssignmentExpression ||
+          (assignmentExpression.left.type !== AST_NODE_TYPES.Identifier &&
+            assignmentExpression.left.type !== AST_NODE_TYPES.MemberExpression)
```

Comment:
(optional) could reuse the utility for this

```suggestion
          !isIdentifierOrMemberOrChainExpression(assignmentExpression.left)
```

The naming is a little awkward but the semantic meaning of that function is maybe something like `isMemberAccessLike` so it's appropriate to reuse here IMO 🤷